crypto/fips1402/src/main/java/org/keycloak/crypto/fips/KeycloakFipsSecurityProvider.java

```diff
@@ -38,23 +36,4 @@ public synchronized final Service getService(String type, String algorithm) {
             return null;
         }
     }
-
-    public static String isSystemFipsEnabled() {
```

Comment:
@mposolda @rmartinc This code does not work in JDK 17 because `java.security.SystemConfigurator` is not there.  Looks like it makes more sense to remove it and move forward without checking if FIPS is enabled to the JVM.

I did not find any alternative in JDK 17. If you have an alternative, that would be nice.

Reply:
@pedroigor Some correction to this as I've already investigated something similar few months ago once you introduced this. That's also why I've slightly changed this code in this commit https://github.com/keycloak/keycloak/commit/4f068fcd  to return 3 statuses (`enabled`, `disabled`, `unknown` ) instead of 2 statuses (`enabled`, `disabled`), which you originally had.

This code actually works even on JDK 17. But strangely enough, it does not work on all JVM platforms. Few details:
- It does not work on my Ubuntu laptop at all. On both OpenJDK 11 and OpenJDK 17 I have `java.lang.ClassNotFoundException: java.security.SystemConfigurator`

- However on RHEL 9, it works fine with OpenJDK 17! When running Keycloak server on this OpenJDK 17 version:
```
openjdk version "17.0.6" 2023-01-17 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.6.0.10-3.el9_1) (build 17.0.6+10-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.6.0.10-3.el9_1) (build 17.0.6+10-LTS, mixed mode, sharing)
```
The code actually works and it is able to correctly detect status. See this, which I've tried this morning:
```
2023-03-27 09:35:48,159 TRACE [org.keycloak.common.crypto.CryptoIntegration] (main) Java security providers: [ 
 KC(BCFIPS version 1.000203, FIPS-JVM: enabled) version 1.0 - class org.keycloak.crypto.fips.KeycloakFipsSecurityProvider, 
 BCFIPS version 1.000203 - class org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider, 
 BCJSSE version 1.0014 - class org.bouncycastle.jsse.provider.BouncyCastleJsseProvider, 
 SunPKCS11-NSS-FIPS version 17 - class sun.security.pkcs11.SunPKCS11, 
 SUN version 17 - class sun.security.provider.Sun, 
 SunEC version 17 - class sun.security.ec.SunEC, 
 SunJSSE version 17 - class sun.security.ssl.SunJSSE, 
 SunJCE version 17 - class com.sun.crypto.provider.SunJCE, 
 SunRsaSign version 17 - class sun.security.rsa.SunRsaSign, 
 XMLDSig version 17 - class org.jcp.xml.dsig.internal.dom.XMLDSigRI, 
]
```

- One interesting aspect: This works when I execute the Keycloak with Quarkus distribution. Which can be done when I test manually on RHEL VM or when running tests as described here: https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/HOW-TO-RUN.md#integration-tests . However when I run the tests as unit tests with Java 17 as described here https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/HOW-TO-RUN.md#unit-tests, then I have also status `unknown`. But the reason is `java.lang.reflect.InaccessibleObjectException: Unable to make static boolean java.security.SystemConfigurator.isSystemFipsEnabled() accessible: module java.base does not "opens java.security" to unnamed module @22f71333` . So the reason is different than on Ubuntu when the class SystemConfigurator is not available at all. This can be good to investigate, but IMO it very low priority as it affects only unit tests.

- I am not sure about Fedora and not sure if it works there or not

So in shortcut: My vote is definitely to keep this code as it works on OpenJDK 17 at least on some platforms like RHEL 9 and UBI 9, which are most important platforms for us and for our customers. The fact that it doesn't work everywhere is OK as long as status `unknown` is reported.